### ISSUE TITLE
consult-outline heading ordering

### ISSUE THREAD

**Issue by linwaytin** [minad/consult#245]
Hi,

Thanks for this fabulous package. The effect of `consult-outline` is better than the corresponding function in `ivy` IMO.

I'm curious if there is a way to force the list shown by `consult-outline` ordered by the line number, that is, following the original ordering.
Sometimes I have headings with similar titles so the ordering can help me find out what I really want.

Thanks.

**Comment by minad** [comment]
There is only one scenario afaik where the ordering is changed. This happens when you use the Emacs built-in `flex` completion style in combination with Icomplete. In that case it is not possible to preserve the sorting without doing some advice-based hacks. If you are indeed using that completion style, I recommend that you use the excellent `orderless` package instead which provides a much more flexible completion style.

**Comment by oantolin** [comment]
> This happens when you use the Emacs built-in `flex` completion style in combination with Icomplete.

Very minor correction: it happens if you use `flex` whether or not you combine it with `icomplete`.

**Comment by linwaytin** [comment]
I didn't use `selectrum` and I think that the reason.
The problem is fixed by `selectrum`. Thank you.

**Comment by oantolin** [comment]
Out of curiosity, what were you using instead of Selectrum, @linwaytin? I've tried with Icomplete, with the default `*Completions*` buffer and with embark-collect and the order seems to be preserved in all of them.

**Comment by oantolin** [comment]
@minad I missed that Selectrum is immune to `flex`'s shenanigans, I should have said, as you did, `flex`+non-Selectrum.

I go through stages in my completion preferences. Right now I'm into minimalism and am using just default completion most of the time (I did rebind `RET` to a command that selects the top completion, default completion is a little annoying without it). I'll sometimes popup the completions when I feel I need them, but I think that 90% or more of the time I don't, specially now that I use consult for so many things and the preview lets me know if I have the right candidate! If I ever grow out of this phase and go back to showing completions by default, I'll use Selectrum. I did try using Selectrum with the one-line display, and that's pretty nice. Still very mininal, but I can visually confirm that I have the right candidate before pressing `RET` in every case, not just when I have consult previews.
